### PR TITLE
Move vuex hdWallet methods to SDK wallet lib

### PR DESCRIPTION
**Pull request by peronczyk**

### CODE REVIEW

File: src/composables/permissions.ts [hunk 1]
```diff
@@ -149,17 +156,28 @@ export function usePermissions() {
...
 
+      // For not recognized transactions we are displaying raw sign modal.
+      // The popupHandler has it's own logic for this.
+      if (
+        method === METHODS.sign
+        && (!modalProps.txBase64 || !isTxOfASupportedType(modalProps.txBase64))
+      ) {
+        modal = MODAL_CONFIRM_RAW_SIGN;
+        popup = POPUP_TYPE_SIGN;
+      }
```
Comment (peronczyk) [inline src/composables/permissions.ts:175]: Previously the `openPopup` was detecting if the transaction is supported by the app or not. The same check was done in the previous permissions Vuex module. So there was 2 places doing the same logic. I consolidated this and moved here as the permissions composable is the only place where the `openPopup` function is fired.

File: src/protocols/aeternity/libs/AeSdkSuperhero.ts [hunk 2]
```diff
@@ -29,15 +29,15 @@ type ISpendOptions = Omit<Parameters<typeof spend>[2], 'onAccount' | 'onNode'>
...
 
-  constructor(store: Store<any>, options: any) {
+  constructor(options: any, nodeNetworkId: Ref<string | undefined>) {
     super(options);
-    this.store = store;
+    this.account = new AeAccountHdWallet(nodeNetworkId);
   }
 
   _resolveAccount() {
-    return new AccountSuperhero(this.store);
```
Comment (peronczyk) [inline src/protocols/aeternity/libs/AeSdkSuperhero.ts:40]: Every composable that had some kind of data polling and was using the SDK was causing to fire this method again and again. So the `AccountSuperhero` was instantiated dozen times per minute. Now it's cached.

File: src/composables/permissions.ts [hunk 3]
```diff
@@ -149,17 +157,28 @@ export function usePermissions() {
...
 
+      // For not recognized transactions we are displaying raw sign modal.
+      // The popupHandler has it's own logic for this.
+      if (
+        method === METHODS.sign
+        && (!modalProps.txBase64 || !isTxOfASupportedType(modalProps.txBase64))
+      ) {
+        modal = MODAL_CONFIRM_RAW_SIGN;
+        popup = POPUP_TYPE_RAW_SIGN;
+      }
```
Comment (peronczyk) [inline src/composables/permissions.ts:176]: Previously the `openPopup` was detecting if the transaction is supported by the app or not. The same check was done in the previous permissions Vuex module. So there was 2 places doing the same logic. I consolidated this and moved here as the permissions composable is the only place where the `openPopup` function is fired.

File: src/composables/permissions.ts [hunk 4]
```diff
@@ -149,17 +157,28 @@ export function usePermissions() {
...
 
+      // The wallet app is able to recognize and properly display only bunch of transaction types.
+      // For every other we need to display different modal with some warnings.
+      if (
+        method === METHODS.sign
+        && (!modalProps.txBase64 || !isTxOfASupportedType(modalProps.txBase64))
+      ) {
+        modal = MODAL_CONFIRM_RAW_SIGN;
+        popup = POPUP_TYPE_RAW_SIGN;
+      }
```
Comment (peronczyk) [inline src/composables/permissions.ts:176]: Previously the `openPopup` was detecting if the transaction is supported by the app or not. The same check was done in the previous permissions Vuex module. So there was 2 places doing the same logic. I consolidated this and moved here as the permissions composable is the only place where the `openPopup` function is fired.

File: src/popup/components/Modals/ConfirmRawSign.vue [hunk 5]
```diff
@@ -83,11 +83,7 @@ export default defineComponent({
 
     const activeAccount = getLastActiveProtocolAccount(PROTOCOL_AETERNITY);
 
-    const dataAsString = computed(
-      (): string => (typeof popupProps.value?.data === 'string')
-        ? popupProps.value?.data
-        : Buffer.from(popupProps.value?.data as any).toString('hex'),
```
Comment (peronczyk) [inline src/popup/components/Modals/ConfirmRawSign.vue:89]: There is no such a thing as `data` anymore.

File: src/popup/components/Modals/ConfirmRawSign.vue [hunk 6]
```diff
@@ -83,11 +83,7 @@ export default defineComponent({
...
-    );
+    const dataAsString = computed((): string => popupProps.value?.txBase64?.toString() || '');
```
Comment (peronczyk) [inline src/popup/components/Modals/ConfirmRawSign.vue:86]: As there is no such a thing as `data` anymore we were able to skip any conditional checks.

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts
Comment (peronczyk) [inline src/protocols/aeternity/libs/AeAccountHdWallet.ts:1]: This is the most important file of the PR. It consists all the logic we had in the Vuex `hdWallet` module. So this is a core of signing stuff and asking user for permission to do it.

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+      );
+      if (!hasPermission) {
+        throw new RpcRejectedByUserError('Rejected by user');
+      }
+    }
+
+    const encodedTx = decode(txBase64);
+    const signature = await this.sign(
+      Buffer.concat([
+        Buffer.from(this.nodeNetworkId.value!),
```
Comment (martinkaintas) [inline src/protocols/aeternity/libs/AeAccountHdWallet.ts:60]: Would it be safer if we throw an error if `this.nodeNetworkId.value` does not exist?

Comment (peronczyk) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts]: Makes sense. I added a check for the `nodeNetworkId` and an error is thrown when it's missing.

**Comment by Liubov-crypto** [comment]
During my testing I've constantly got this issue in console, and I think it might be affect all errors that I listed below:

Error: pubkey cannot be null or undefined.

err1

1. Can't send ae and Fungible tokens: 
2. Verify an amount and Max button isn't working for FT:

https://github.com/superhero-com/superhero-wallet/assets/69896204/d5f810e3-e958-43ed-a1cb-6ff8d2b44828

3. Set name pointer: 
set pointer

4. Register a name:
reg

5. Multisig creation
6. Generate Gift card
7. Bid on name
8. Transactions on DEX:
swap
9. Interactions with superhero.com  - I can connect my wallet but there is no popup for signing Txs.

@peronczyk please check it.

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts
Comment (davidyuk) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts:1]: This class looks good to me 👍

**Comment by peronczyk** [comment]
@Liubov-crypto thank you for your comments. I addressed your feedback. Please retest.

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+    const encodedTx = decode(txBase64);
+    const signature = await this.sign(
+      Buffer.concat([
+        Buffer.from(this.nodeNetworkId.value),
+        Buffer.from(encodedTx),
+      ]),
+      options, // Mainly to pass the `fromAccount` property
+    );
+    // @ts-ignore
```
Comment (martinkaintas) [inline src/protocols/aeternity/libs/AeAccountHdWallet.ts:63]: I suggest to replace this with `// @ts-expect-error` and to specify why it is needed

**Review by martinkaintas — APPROVED** [review]
lgtm

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts (hunk 8 above)
Comment (peronczyk) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts]: It turns out I can remove this :) So cool to see the code just fit as expected.

**Review by CedrikNikita — APPROVED** [review]
Code changes look reasonable.

**Comment by subhod-i** [comment]
Code changes looks good to me. Apart of that please fix the bugs we discussed.

File: src/protocols/aeternity/libs/AeAccountHdWallet.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+  override async signMessage(
+    message: string,
+    options: Parameters<AccountBase['signMessage']>[1],
+  ): Promise<Uint8Array> {
+    if (options?.aeppOrigin) {
+      const { checkOrAskPermission } = usePermissions();
+      const permissionGranted = await checkOrAskPermission(
+        options.aeppOrigin,
+        METHODS.signMessage,
+        { message },
```
[the remark this answers is not in the document]
Comment (peronczyk) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts:75]: Yes, it's important, but it's one of the possible options for the `IModalProps`.  Function `checkOrAskPermission` receives the 3rd argument as a list of things that will be passed to the modal/popup so this data can be displayed to user to let him know what he is actually signing. Do you have any suggestions on how to change this?

[the remark this answers is not in the document]
Comment (subhod-i) [reply src/protocols/aeternity/libs/AeAccountHdWallet.ts:75]: No suggestions, I thought you might have missed it. Resolving.

**Review by subhod-i — APPROVED** [review]
Cool PR.

**Pull request merged**